### ISSUE TITLE
rspec/autorun doesn't honor expose_dsl_globally default

### ISSUE THREAD

**Issue by zeha** [rspec/rspec-core#1931]
The ChangeLog claims expose_dsl_globally would default to true, but this simple test doesn't work:

```ruby
require 'rspec/autorun'
describe 1 do
  it "is < 2" do
    expect(1).to be < 2
  end

  it "has an intentional failure" do
    expect(1).to be > 2
  end
end
```

```shell
% ruby foo.rb
foo.rb:3:in `<main>': undefined method `describe' for main:Object (NoMethodError)
```

If I add this line after `require 'rspec/autorun'`, it's working as expected:
```
RSpec.configure { |c| c.expose_dsl_globally = true }
```

**Comment by myronmarston** [comment]
Thanks for reporting this.  I figured out the cause of this.

The DSL is defaulted to globally exposed when the config object is instantiated:

https://github.com/rspec/rspec-core/blob/978f0b7edc33ed157b052eddfdbd87e625dadccb/lib/rspec/core.rb#L86

When you run things via `rspec`, it access `RSpec.configuration` before loading your code.  When you use `rspec/autorun`, it doesn't hit `RSpec.configuration` before hitting your code, so the DSL hasn't been exposed globally.

For now, you can just add `RSpec.configuration` after `require 'rspec/autorun'` (accessing config is sufficient; not need to set `expose_dsl_globally = true`) but I plan to work up a fix to address this.

**Comment by myronmarston** [comment]
Addressed by #1933.
